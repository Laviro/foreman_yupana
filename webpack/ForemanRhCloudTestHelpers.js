export const inventoryStateWrapper = innerState => ({
  ForemanRhCloud: {
    inventoryUpload: { ...innerState },
  },
});
