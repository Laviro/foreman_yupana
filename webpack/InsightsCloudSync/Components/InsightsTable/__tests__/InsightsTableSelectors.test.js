import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';
import { insightsStateWrapper } from '../../../../ForemanRhCloudTestHelpers';
import { routerState, APIState, APIErrorState } from './fixtures';
import {
  selectError,
  selectHits,
  selectInsightsCloudTable,
  selectIsAllSelected,
  selectItemCount,
  selectPage,
  selectPerPage,
  selectQuery,
  selectQueryParams,
  selectSearch,
  selectSelectedIds,
  selectShowSelectAllAlert,
  selectSortBy,
  selectSortOrder,
  selectStatus,
} from '../InsightsTableSelectors';

const state = {
  ...routerState,
  ...APIState,
  ...insightsStateWrapper({
    table: {
      selectedIds: {
        '51': true,
      },
      showSelectAllAlert: true,
      isAllSelected: false,
    },
  }),
};

const fixtures = {
  'should return router query': () => selectQuery(state),
  'should return router search': () => selectSearch(state),
  'should return router page': () => selectPage(state),
  'should return router perPage': () => selectPerPage(state),
  'should return router sort by': () => selectSortBy(state),
  'should return router sort order': () => selectSortOrder(state),
  'should return queryParams': () => selectQueryParams(state),
  'should return API status': () => selectStatus(state),
  'should return API error': () => selectError({ ...state, ...APIErrorState }),
  'should return API hits': () => selectHits(state),
  'should return API item count': () => selectItemCount(state),
  'should return insights table': () => selectInsightsCloudTable(state),
  'should return insights selectedIds': () => selectSelectedIds(state),
  'should return insights isAllSelected': () => selectIsAllSelected(state),
  'should return insights showSelectAllAlert': () =>
    selectShowSelectAllAlert(state),
};

describe('InsightsTable selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
