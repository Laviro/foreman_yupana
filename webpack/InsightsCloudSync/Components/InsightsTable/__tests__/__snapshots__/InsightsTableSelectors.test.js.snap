// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsightsTable selectors should return API error 1`] = `"Request failed with status code 503"`;

exports[`InsightsTable selectors should return API hits 1`] = `
Array [
  Object {
    "host_id": 1,
    "hostname": "foo.example.com",
    "id": 16,
    "last_seen": "2020-11-01T15:55:14.280+02:00",
    "likelihood": 1,
    "publish_date": "2017-11-02T14:00:00.000+02:00",
    "results_url": "https://cloud.redhat.com/insights/advisor/recommendations/hardening_yum%7CHARDENING_YUM_GPG_3RD_6/716306d2-b4f8-446f-82c8-a45af30110bd/",
    "solution_url": "",
    "title": "Decreased security: Yum GPG verification disabled",
    "total_risk": 1,
  },
  Object {
    "host_id": 1,
    "hostname": "foo.example.com",
    "id": 17,
    "last_seen": "2020-11-01T15:55:14.280+02:00",
    "likelihood": 3,
    "publish_date": "2019-07-12T02:08:00.000+03:00",
    "results_url": "https://cloud.redhat.com/insights/advisor/recommendations/packages_across_major_release%7CPACKAGES_ACROSS_MAJOR_RELEASE/716306d2-b4f8-446f-82c8-a45af30110bd/",
    "solution_url": "https://access.redhat.com/node/54483",
    "title": "Installation of packages across major releases is not supported",
    "total_risk": 2,
  },
]
`;

exports[`InsightsTable selectors should return API item count 1`] = `2`;

exports[`InsightsTable selectors should return API status 1`] = `"RESOLVED"`;

exports[`InsightsTable selectors should return insights isAllSelected 1`] = `false`;

exports[`InsightsTable selectors should return insights selectedIds 1`] = `
Object {
  "51": true,
}
`;

exports[`InsightsTable selectors should return insights showSelectAllAlert 1`] = `true`;

exports[`InsightsTable selectors should return insights table 1`] = `
Object {
  "isAllSelected": false,
  "selectedIds": Object {
    "51": true,
  },
  "showSelectAllAlert": true,
}
`;

exports[`InsightsTable selectors should return queryParams 1`] = `
Object {
  "isSelectAll": false,
  "page": 1,
  "perPage": 7,
  "query": "total_risk < 3",
  "sortBy": "total_risk",
  "sortOrder": "asc",
}
`;

exports[`InsightsTable selectors should return router page 1`] = `1`;

exports[`InsightsTable selectors should return router perPage 1`] = `7`;

exports[`InsightsTable selectors should return router query 1`] = `
Object {
  "page": "1",
  "per_page": "7",
  "search": "total_risk+%3C+3",
  "sort_by": "total_risk",
  "sort_order": "asc",
}
`;

exports[`InsightsTable selectors should return router search 1`] = `"total_risk < 3"`;

exports[`InsightsTable selectors should return router sort by 1`] = `"total_risk"`;

exports[`InsightsTable selectors should return router sort order 1`] = `"asc"`;
